public/apps/configuration/panels/permission-list/edit-modal.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiForm,
+  EuiFieldText,
+  EuiComboBox,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { ComboBoxOptions, Action } from '../../types';
+import { stringToComboBoxOption, comboBoxOptionToString } from '../../utils/combo-box-utils';
+import { FormRow } from '../../utils/form-row';
+
+interface PermissionEditModalDeps {
+  groupName: string;
+  action: Action;
+  allowedActions: string[];
+  optionUniverse: ComboBoxOptions;
+  handleClose: () => void;
+  handleSave: (groupName: string, allowedActions: string[]) => Promise<void>;
+}
+
+const TITLE_DICT: { [key: string]: string } = {
+  [Action.create]: 'Create new action group',
+  [Action.edit]: 'Edit action group',
+  [Action.duplicate]: 'Duplicate action group',
+};
+
+export function PermissionEditModal(props: PermissionEditModalDeps) {
+  const [groupName, setGroupName] = useState<string>(props.groupName);
+  const [allowedActions, setAllowedActions] = useState<ComboBoxOptions>(
+    props.allowedActions.map(stringToComboBoxOption)
+  );
+
+  const handleGroupNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+    if (newValue.length <= 50) {
```

Comment:
Is 50 from UX requirement or some system limitation? Better to have a const for better context. Probably we can consider some error message in case the input has length over 50.

Reply:
It is in the help text (see line 74)